src/cosmoz-toggle.js

```diff
@@ -0,0 +1,34 @@
+import { html, component } from 'haunted';
+import { toggleStyles } from '../src/toggle-styles.ts';
+
+export const CosmozToggle = (host) => {
+	const { label, checked, onChange } = host;
+	const onChecked = (event) => {
+		onChange?.(event.target.checked);
+		host.dispatchEvent(
+			new CustomEvent('change', { detail: { value: event.target.checked } })
+		);
+	};
+
+	return html`
+		<style>
+			${toggleStyles}
+		</style>
+		<div class="container">
```

Comment:
You can drop the container and use the host as the container. This enables easier styling.